src/ChargeClustering/ChargeClustering.jl

```diff
@@ -8,31 +8,38 @@ function cluster_detector_hits(
     detno::AbstractVector{<:Integer},
     edep::AbstractVector{TT},
     pos::AbstractVector{<:Union{StaticVector{3,PT}, CartesianPoint{PT}}},
-    cluster_radius::RealQuantity
-) where {TT<:RealQuantity, PT <: RealQuantity}
+    thit::AbstractVector{TTT},
+    cluster_radius::RealQuantity,
+    cluster_time::RealQuantity
+) where {TT<:RealQuantity, PT <: RealQuantity, TTT <: RealQuantity}
 
     pos = isa(first(pos), CartesianPoint) ? [SVector(p.x, p.y, p.z) for p in pos] : pos     #converting CartesianPoint to SVector
     pos = [SVector(p.x, p.y, p.z) for p in pos]
-
+    
     Table = TypedTables.Table
-    unsorted = Table(detno = detno, edep = edep, pos = pos)
+    unsorted = Table(detno = detno, edep = edep, pos = pos, thit = thit)
     sorting_idxs = sortperm(unsorted.detno)
     sorted = unsorted[sorting_idxs]
     grouped = Table(consgroupedview(sorted.detno, TypedTables.columns(sorted)))
 
     r_detno = similar(detno, 0)
     r_edep = similar(edep, 0)
     r_pos = similar(pos, 0)
+    r_thit = similar(thit, 0)
 
     posunit = unit(PT)
     ustripped_cradius = ustrip(posunit, cluster_radius)
+
+    thitunit = unit(TTT)
+    ustripped_ctime = ustrip(thitunit, cluster_time)
     
     for d_hits_nt in grouped
         d_hits = Table(d_hits_nt)
         d_detno = first(d_hits.detno)
         @assert all(isequal(d_detno), d_hits.detno)
         if length(d_hits) > 3
-            clusters = Clustering.dbscan(ustrip.(flatview(d_hits.pos)), ustripped_cradius, leafsize = 20, min_neighbors = 1, min_cluster_size = 1)
+            clusters = st_dbscan(hcat(ustrip.(d_hits.pos)...)', ustrip.(d_hits.thit), eps_space = ustripped_cradius, eps_time = ustripped_ctime, min_cluster_size = 1)

```

Comment:
Maybe instead of converting everything first to an `SVector` and then to this `Matrix` for `st_dbscan`, we might consider defining a function that defines this matrix directly from `grouped` or `d_hits.pos`

Could be something like (I did not test this, just wrote this out in this comment as a starting point):
```julia
function create_dbscan_matrix(pos::AbstractVector{CartesianPoint{T}})
    ustrip.(hcat(getfield.(pos, :x), getfield.(pos, :y), getfield.(pos, :z)))'
end

function create_dbscan_matrix(pos::AbstractVector{StaticVector{T}})
    ustrip.(hcat(d_hits.pos)...))'
end